### ISSUE TITLE
--keychain argument not necessary

### ISSUE THREAD

**Issue by maccman** [electron/electron-notarize#88]
Can we get rid of it. Or at least make it optional. notarytool will just use the default keychain.

**Comment by aguynamedben** [comment]
@maccman What's up! Doing this too.

I agree it's unnecessary per https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow, only `--keychain-profile` seems necessary.

Did you find a workaround? I tried passing `Local Items` and `login` but no dice...
```
configs/notarize.js: Notarization failed : Failed to notarize via notarytool

Error: The value 'login' is invalid for '--keychain <keychain>': The file couldn’t be opened because it doesn’t exist.
Usage: notarytool submit <options>
  See 'notarytool submit --help' for more information.
```

How we're calling it:
```js
return notarize({
  tool: 'notarytool',
  appPath: `${appOutDir}/${appName}.app`,
  keychain: 'login',
  keychainProfile: 'AC_PASSWORD',
}).catch((error) => {
  console.error(`configs/notarize.js: Notarization failed : ${error.message}`);
  process.exit(2);
});
```

Thank you @MarshallOfSound for adding `notarytool` support 🙏  It will be so nice to not have to install Xcode. I think `altool` was the only reason I ever had to install Xcode.

**Comment by aguynamedben** [comment]
Might have PR incoming that supports one or both... testing...

**Comment by aguynamedben** [comment]
FYI docs for this command can be seen with

```
$ xcrun notarytool submit --help
OVERVIEW: Submit an archive to the Notary service

USAGE: notarytool submit <options>

ARGUMENTS:
  <file-path>             Path to the archive

OPTIONS:
  -v, --verbose
  -k, --key <key>         App Store Connect API key. File system path to the private key.
  -d, --key-id <key-id>   App Store Connect API Key ID. Usually 10 alphanumeric characters.
  -i, --issuer <issuer>   App Store Connect API Issuer ID. UUID format.
  --apple-id <apple-id>   Developer Apple ID.
  --password <password>   App-specific password for your Apple ID. You will be given a secure prompt on the command line if Apple ID and Team ID are provided and '--password' option is not specified.
  --team-id <team-id>     Developer Team ID.
  -p, --keychain-profile <keychain-profile>
                          Authenticate with credentials stored in the Keychain. Use the profile name you provided in the "store-credentials" command.
  --keychain <keychain>   Pass the path to a keychain file to use for reading the keychain item. If the specified keychain is locked, you'll be prompted to unlock it.
  -f, --output-format <output-format>
                          Desired output format. Note that 'json' and 'plist' are incompatible with '--progress'; a single update will be output at the end of the operation. Choices: ["normal", "json", "plist"] (default: normal)
  --progress/--no-progress
                          Display progress indicators. Only compatible with normal output format (default) and will be otherwise suppressed. (default: true)
  --webhook <webhook>     Designate a public webhook URL to receive HTTP notifications on.
  --wait/--no-wait        Wait until processing is complete. Optionally set a '--timeout' if waiting. (default: false)
  --timeout <duration>    Optional time limit for 'wait'. notarytool will exit after polling for <duration>. The Notary service will continue processing even if the timeout is reached.
        'duration' is an integer followed by an optional suffix: seconds 's' (default), minutes 'm', hours 'h'. Examples: '3600', '60m', '1h'
  --s3-acceleration/--no-s3-acceleration
                          Use S3 Transfer Acceleration for uploads. (default: true)
  --version               Show the version.
  -h, --help              Show help information.
```

**Comment by aguynamedben** [comment]
Hmm, with this diff:
<img width="678" alt="2021-11-22_17-38-39" src="https://user-images.githubusercontent.com/107841/142959490-9632f0d4-ef66-4a8c-9c98-9d5ba25111c3.png">

Notarization gives this error:
```
configs/notarize.js: Notarization failed : The keychain property is required when using notarization with keychain credentials
```

So I guess I need to figure out how to set the value properly 😓 This is the command I used to set my app-specific password:
```
$ xcrun notarytool store-credentials "AC_PASSWORD" --apple-id "myemailaddress@gmail.com" --team-id "RXXXXXXXXX" --password "foos-bars-bazz-bang"
```

**Comment by MarshallOfSound** [comment]
`security list-keychains` gives you the possible values for keychains on your system, normally they are absolute or user-specific paths.

**Comment by aguynamedben** [comment]
## Can't find a valid value for `keychain`
Thank you for that tip on how to list keychains. Here's what I get:
```
$ security list-keychains
    "/Users/standefer/Library/Keychains/login.keychain-db"
    "/Library/Keychains/System.keychain"
```
With electron-notarize 1.1.1 installed (no modifications), I'm running this:
<img width="577" alt="2021-11-22_21-54-01" src="https://user-images.githubusercontent.com/107841/142977007-7a232259-f7a8-4a7d-93af-c0cb1629d46b.png">
but get this:
<img width="960" alt="2021-11-22_21-59-59" src="https://user-images.githubusercontent.com/107841/142977452-92ef6963-09d3-44da-a878-f08779f80b55.png">
This happens if I use the other path, just the last part of the path, etc. I can't figure any way to get the `--keychain` argument to work. 🤷‍♂️

## Notarization works without `keychain`, but staple fails

I was able to run notarization with my AC_PASSWORD keychain entry described above (and without `--keychain` as the Apple docs show) and it succeeded. The command was:
```
$ xcrun notarytool submit release/Command\ E-1.31.0-alpha.2-mac.zip --keychain-profile "AC_PASSWORD"
```

Here's AC_PASSWORD was setup in my Keychain Access:
<img width="995" alt="2021-11-22_21-46-26" src="https://user-images.githubusercontent.com/107841/142976528-bdb73505-6acb-49d1-80c3-ab221ee47f9b.png">

I modified electron-notarize to not require `--keychain` ([PR](https://github.com/electron/electron-notarize/pull/91)) and ran it with debug on like this:
```
$ DEBUG=* yarn release
```

When electron-builder got the electron-notarize step, I could see the following output showing that notarization worked without `--profile`, but the subsequent "staple" step failed with Error 65.
<img width="1314" alt="2021-11-22_21-39-51" src="https://user-images.githubusercontent.com/107841/142976671-14ec4fec-ccc2-4aff-882b-89d9e0580d3a.png">

Here's Error 65
<img width="905" alt="2021-11-22_21-43-49" src="https://user-images.githubusercontent.com/107841/142976684-eac743d0-25f3-4541-8f2d-1a1f67d4b0bc.png">

I've never seen the Error 65 around stapling, but I saw [this ticket](https://github.com/electron/electron-notarize/issues/31) and am working on a PR to retry stapling to see if it's just a race condition with Apple servers.

Let me know if you have any other ideas. Thanks!

**Comment by aguynamedben** [comment]
Welp, I'm not sure what state my local `xcrun notarytool` keeps going between, but now I've finished downloading Xcode 13.1 (backup option, us `altool`) and agreed to the license and now notarytool is back to telling me `keychain` is required. 🤷‍♂️ 😮‍💨

**Comment by maccman** [comment]
I'll leave a comment as soon as I'm back at my computer￼

**Comment by jwatte** [comment]
Try this instead: https://github.com/electron/electron-notarize/pull/94

**Comment by MarshallOfSound** [comment]
Closed by #94

**Comment by tlebon** [comment]
in case anyone comes here like i did, I just reran the command to add a credential `xcrun notarytool store-credential` and pointed at the login-keychain location. 
find this by running `security list-keychains`
and then append to the end with `--keychain <keychain-location>`
